src/System.Data.SQLite/NativeMethods.cs

```diff
@@ -127,6 +127,56 @@ internal static class NativeMethods
 		[DllImport(c_dllName, CallingConvention = c_callingConvention)]
 		public static extern int sqlite3_trace_v2(SqliteDatabaseHandle db, SQLiteTraceEvents eventsMask, SQLiteTraceV2Callback callback, IntPtr userData);
 
+#if NET5_0
+		static NativeMethods()
+		{
+			var assembly = typeof(NativeMethods).Assembly;
+			var configFile = System.IO.Path.GetFileName(assembly.Location) + ".config";
+			if (System.IO.File.Exists(configFile))
+			{
+				try
+				{
+					using var stream = new System.IO.FileStream(configFile, System.IO.FileMode.Open);
+					using var reader = System.Xml.XmlReader.Create(stream);
+					var dllMapConfig = System.Xml.Linq.XDocument.Load(reader);
+
+					
+					foreach (var element in dllMapConfig.Root.Elements())
+					{
+						if (element.Name != "dllmap")
+							continue;
+
+						if (element.Attribute("dll").Value == c_dllName)
+						{
+							s_mappedDllName = element.Attribute("target").Value;
+							break;
+						}
+					}
+
+					if (!string.IsNullOrEmpty(s_mappedDllName))
+						NativeLibrary.SetDllImportResolver(typeof(NativeMethods).Assembly, DllImportResolver);
+				}
+				catch (Exception)
+				{
+				}
+			}
+		}
+
+		private static IntPtr DllImportResolver(string libraryName, System.Reflection.Assembly assembly, DllImportSearchPath? searchPath)
+		{
+			var handle = IntPtr.Zero;
+
+			if (libraryName == c_dllName && !string.IsNullOrEmpty(s_mappedDllName))
+				NativeLibrary.TryLoad(s_mappedDllName, out handle);
+
+			System.Console.Error.WriteLine($"DllImportResolver for {libraryName} {handle}");
+
+			return handle;
+		}
+
+		private static string s_mappedDllName;
+#endif
+
 		const string c_dllName = "sqlite3";
```

Comment:
Instead of shipping a (hardcoded?) `System.Data.SQLite.dll.config` file that has a `target` specifying the actual DLL name... can we just change this line?

Reply:
I believe it needs to be different on Windows vs. Mac and there's no way to differentiate with a string constant unless Windows and Mac use a different target framework, which they don't currently and may not be able to.

Reply:
So the Mac consumer will copy in an external XML file with this exact syntax? (Because it can't go inside the package, otherwise we're back to the same problem.) Is there any other code that uses `System.Data.SQLite.dll.config`? If we have to process it manually... could we just have `System.Data.SQLite.dll.config.txt` and use `File.ReadAllText` to read a single line from it, which is the dylib name to use?

Reply:
> So the Mac consumer will copy in an external XML file with this exact syntax? (Because it can't go inside the package, otherwise we're back to the same problem.)  Is there any other code that uses System.Data.SQLite.dll.config? If we have to process it manually... could we just have System.Data.SQLite.dll.config.txt and use File.ReadAllText to read a single line from it, which is the dylib name to use?

This is they way Mono worked, the syntax is defined by that for historical purposes. Using the existing `.dll.config` file that gets put there by the `Libronix.Utility.Sqlite` allows for one file to continue to be used by both net5.0+ and full framework.

The file is part of `Libronix.Utility.Sqlite` (or really the consuming app if so desired) to allow targeting a different native assembly version which is distributed by the `Libronix.Utility.Sqlite` package.